Eatery/Views/Eateries/PopularTimesView.swift

```diff
@@ -0,0 +1,205 @@
+
+            if self.isExpanded {
+                self.histogramCollapseConstraint?.deactivate()
+                self.histogramExpandConstraint?.activate()
+            } else {
+                self.histogramExpandConstraint?.deactivate()
+                self.histogramCollapseConstraint?.activate()
+            }
+
+            self.layoutDelegate?.popularTimesContentSizeDidChange(self)
+        }
+
+        if animated {
+            UIViewPropertyAnimator(duration: 0.5, dampingRatio: 1.0, animations: actions).startAnimation()
+        } else {
+            actions()
+        }
+    }
+
+}
+
+extension PopularTimesView: HistogramViewDataSource {
+
+    // Overflowed military hour is a time between 0...47
+    private func containOverflowedMilitaryHour(hour: Int) -> Int {
+        return (hour >= 24) ? hour - 24 : hour
+    }
+
+    private func formattedHourForTime(militaryHour: Int) -> String {
+        var formattedHour: String!
+        if militaryHour > 12 {
+            formattedHour = "\(militaryHour - 12)p"
+        } else if militaryHour == 12 {
+            formattedHour = "12p"
+        } else if militaryHour > 0 {
+            formattedHour = "\(militaryHour)a"
+        } else {
+            formattedHour = "12a"
+        }
+        return formattedHour
+    }
+
+    func numberOfDataPoints(for histogramView: HistogramView) -> Int {
+        return overflowedEndHour - startHour
+    }
+
+    func histogramView(_ histogramView: HistogramView, relativeValueOfDataPointAt index: Int) -> Double {
+        return eatery.swipeDensity(for: containOverflowedMilitaryHour(hour: startHour + index))
+    }
+
+    func histogramView(_ histogramView: HistogramView, descriptionForDataPointAt index: Int) -> NSAttributedString? {
+        let currentHour = Calendar.current.component(.hour, from: Date())
+        let barHour = containOverflowedMilitaryHour(hour: startHour + index)
+
+        let hourText = currentHour == barHour ? "Now: " : formattedHourForTime(militaryHour: barHour).appending(": ")
+
+        let waitTimesText: String
+        if let (low: lowEstimate, high: highEstimate) = eatery.waitTimes(atHour: barHour, minute: 0) {
+            waitTimesText = "\(lowEstimate)-\(highEstimate)m"
+        } else {
+            waitTimesText = "?"
+        }
+
+        let labelText = "\(hourText)\(waitTimesText) wait"
+
+        let string = NSMutableAttributedString(
+            string: labelText,
+            attributes: [NSAttributedStringKey.font: UIFont.systemFont(ofSize: 14, weight: .medium)]
+        )
+        string.addAttributes(
+            [NSAttributedStringKey.foregroundColor: UIColor.eateryBlue,
+             NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 14)
+            ],
+            range: NSRange(location: hourText.count, length: waitTimesText.count)
+        )
+        return string
+    }
+
+    func numberOfDataPointsPerTickMark(for histogramView: HistogramView) -> Int? {
+        return 3
```

Comment:
can we add comment about why this is 3?

Reply:
I can't think of any comment that isn't a restatement of the method name?